taichi/transforms/scalarize.cpp

```diff
@@ -270,12 +278,12 @@ class Scalarize : public BasicStmtVisitor {
     auto const &formats = stmt->formats;
     std::vector<std::variant<Stmt *, std::string>> new_contents;
     // Sparse mapping between formatted expr and its specifier
-    std::map<Stmt *, std::string> new_formats;
+    std::vector<std::optional<std::string>> new_formats;
     for (size_t i = 0; i < contents.size(); i++) {
```

Comment:
```suggestion
    for (size_t content_i = 0; content_i < contents.size(); content_i++) {
```
This `i` was shallowed by other index variables defined in the inner for loop, sorry for introducing this bug